plugins/DigitalLibrary/tests/TestDigitalLibraryApiClient.py

```diff
@@ -0,0 +1,87 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from cura.CuraApplication import CuraApplication
+from src.DigitalFactoryApiClient import DigitalFactoryApiClient
+from src.PaginationManager import PaginationManager
+
+
+@pytest.fixture
+def application():
+    app = MagicMock(spec=CuraApplication, name = "Mocked Cura Application")
+    return app
+
+
+@pytest.fixture
+def pagination_manager():
+    manager = MagicMock(name = "Mocked Pagination Manager")
+    return manager
+
+
+@pytest.fixture
+def api_client(application, pagination_manager):
+    api_client = DigitalFactoryApiClient(application, MagicMock())
+    api_client._projects_pagination_mgr = pagination_manager
+    return api_client
+
+
+def test_getProjectsFirstPage(api_client):
+    # setup
+    http_manager = MagicMock()
+    api_client._http = http_manager
+    pagination_manager = api_client._projects_pagination_mgr
+    pagination_manager.limit = 20
+
+    finished_callback = MagicMock()
+    failed_callback = MagicMock()
+
+    # Call
+    api_client.getProjectsFirstPage(on_finished = finished_callback, failed = failed_callback)
+
+    # Asserts
+    pagination_manager.reset.assert_called_once()  # Should be called since we asked for new set of projects
+    http_manager.get.assert_called_once()
+    args = http_manager.get.call_args_list[0]
+
+    # Ensure that it's called with the right limit
+    assert args[0][0] == 'https://api.ultimaker.com/cura/v1/projects?limit=20'
```

Comment:
Isn't it better to construct this ground truth URL from the API constant, in case it changes so that the test then doesn't break too?

Reply:
I think we only use the API constant to distinguish between production and staging. So in that sense it should always be the api.ultimaker.com url.